lib/rubocop/cop/rspec/repeated_example_group_body.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Check for repeated describe and context block body.
+      #
+      # @example
+      #
+      #    # bad
+      #    describe 'cool feature x' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    describe 'cool feature y' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    # good
+      #    describe 'cool feature' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    describe 'another cool feature' do
+      #      it { another_predicate }
+      #    end
+      #
+      #    # bad
+      #    context 'when case x' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    context 'when case y' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    # good
+      #    context 'when case x' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    context 'when case y' do
+      #      it { another_predicate }
+      #    end
+      #
+      class RepeatedExampleGroupBody < Cop
+        MSG = 'Repeated %<group>s block body on line(s) %<loc>s'
+
+        def_node_matcher :several_blocks?, '(begin #example_group_with_body?+ )'
+
+        def_node_matcher :metadata, '(block (send _ _ _ $...) ...)'
+        def_node_matcher :body, '(block send args $...)'
+
+        def on_begin(node)
+          return unless several_blocks?(node)
+
+          repeated_group_bodies(node).each do |group, repeats|
+            add_offense(group, message: message(group, repeats))
+          end
+        end
+
+        private
+
+        def repeated_group_bodies(node)
+          node
+            .children
+            .group_by { |child| group_signature(child) }
+            .values
+            .reject(&:one?)
+            .map { |groups| add_repeated_lines(groups) }
+            .flatten(1)
+        end
+
+        def group_signature(group)
+          [metadata(group), body(group)]
```

Comment:
Since those two are used together, probably combine `metadata` and `body` matchers, and use two captures in that common matcher?